Sources/Application/MainCoordinator.swift

```diff
@@ -12,8 +12,9 @@ final class MainCoordinator: NSObject,
     var childCoordinators = [ChildCoordinator]()
     var analyticsCenter: AnalyticsCentral
     let userStore: UserStorable
-    let tokenHolder = TokenHolder()
+    let tokenHolder = TokenHolder.shared
     private var tokenVerifier: TokenVerifier
+    static var reauth = false
```

Comment:
I spent some time thinking/researching how this could be done for reauth scenarios where the user is already within the app (LoginCoordinator is not being presented).
I had a look into ObservableObject and posting notifications but this felt like the easiest and cleanest way to do it. Would be happy to change this if there's a more preferable means of achieving this.

Reply:
I'm ok with using a flag, but I think it needs a better name.  requiresReauth or isReauthing or something like that.